docs/manual/linux-udev.md

```diff
@@ -3,19 +3,93 @@ layout: default
 title: Configuring udev on Linux
 parent: User manual
 nav_order: 6
+has_toc: true
 redirect_from:
   - /getting-started/linux-udev.html
 ---
 
-# Configuring udev on Linux
+# Configuring `udev` rules for VIA and Vial on Linux
 
-In order for your keyboard to be detected on Linux, you need to set up a custom udev rule. Open a terminal and execute the following commands to set this up:
+In order for your keyboard to be detected by both VIA and Vial on Linux, you need to set up a custom `udev` rule. Both use the `hidraw` linux driver to directly communicate configurations to your keyboard. In most situations it should be safe to use a generalized `udev` rule for all `hidraw` devices, but in other cases where you prefer or are required to restrict access on a device basis, it is possible to create a `udev` rule specifically for your keyboard.
+
+The following guides will show you how to implement these `udev` rules. You will require root access to create the files necessary.
+
+## Vial `udev` rule
+
+For a generalized access rule for any device with Vial firmware, write this text to `/etc/udev/rules.d/99-vial.rules`:
+
+```
+KERNEL=="hidraw*", SUBSYSTEM=="hidraw", ATTRS{serial}=="vial:f64c2b3c", MODE="0660", GROUP="users", TAG+="uaccess", TAG+="udev-acl"
+```
+
+**OR** run this in your shell (replace `sudo` with `pkexec` or `doas` if needed):
+
+```
+sudo echo "KERNEL==\"hidraw*\", SUBSYSTEM==\"hidraw\", ATTRS{serial}==\"vial:f64c2b3c\", MODE=\"0660\", GROUP=\"users\", TAG+=\"uaccess\", TAG+=\"udev-acl\"" > /etc/udev/rules.d/99-vial.rules
+```
+
+In order for the rule to take effect, you must [reload `udev`](#reloading-udev).
+
+## Generalized VIA `udev` rule
+
+On a kernel level, there is no way to detect if a keyboard is compatible with VIA, so you have to allow access to all `hidraw` devices, or configure it on a [device-specific basis](#device-specific-udev-rules).
+
+For a rule that allows users to access all `hidraw` devices, write this text to `/etc/udev/rules.d/92-viia.rules`:
+
+```
+KERNEL=="hidraw*", SUBSYSTEM=="hidraw", MODE="0660", GROUP="users", TAG+="uaccess", TAG+="udev-acl"
+```
+
+**OR** run this in your shell (replace `sudo` with `pkexec` or `doas` if needed):
+
+```
+sudo echo "KERNEL==\"hidraw*\", SUBSYSTEM==\"hidraw\", MODE=\"0660\", GROUP=\"users\", TAG+=\"uaccess\", TAG+=\"udev-acl\"" > /etc/udev/rules.d/92-viia.rules
```

Comment:
same comments as above regarding shell redirection/udev commands

Reply:
Resolved this in https://github.com/vial-kb/vial-kb.github.io/pull/19/commits/edbd12201e3981c0a999ba9e07321df5e29eae14